terminator/src/platforms/macos.rs

```diff
@@ -151,22 +152,66 @@ impl MacOSEngine {
+        debug!("[macOS][wrap_element] ax_element: {:?}", ax_element);
+
+        // Try to check element validity by first checking if "AXRole" is present in attribute names
+        let mut is_valid = false;
+        // println!("[macOS][wrap_element] ax_element: {:?}", ax_element);
+        // println!("[macOS][wrap_element] ax_element.0: {:?}", ax_element.0);
+        match ax_element.0.attribute_names() {
+            Ok(attr_names) => {
+                // ItemRef<CFString> does not have as_str(), so compare using .to_string()
+                let has_role = attr_names.iter().any(|attr| attr.to_string() == "AXRole");
+                // println!("[macOS][wrap_element] attr_names: {:?}", attr_names);
+                if has_role {
+                    // Try to get the role if the attribute exists
+                    match ax_element.0.role() {
+                        Ok(_) => {
+                            is_valid = true;
+                        }
+                        Err(e) => {
+                            if let accessibility::Error::Ax(code) = e {
+                                if code != -25204 {
+                                    // kAXErrorNoValue
+                                    let err_str = unsafe { error_string(code) };
+                                    debug!(
+                                        "Warning: Potentially invalid AXUIElement: {:?} (error: {})",
+                                        e, err_str
+                                    );
+                                }
+                            } else {
+                                debug!("Warning: Potentially invalid AXUIElement: {:?}", e);
+                            }
+                            is_valid = false;
+                        }
                     }
                 } else {
-                    debug!("Warning: Potentially invalid AXUIElement: {:?}", e);
+                    debug!(
+                        "AXUIElement does not have 'AXRole' attribute. Attributes: {:?}",
+                        attr_names
+                    );
+                    // If it doesn't have AXRole, we still consider it valid for wrapping,
+                    // but log for debugging.
+                    is_valid = false;
                 }
-                false
             }
-        };
+            Err(e) => {
+                let err_str = if let accessibility::Error::Ax(code) = e {
+                    unsafe { error_string(code) }
+                } else {
+                    "<not an AX error>"
+                };
+                debug!(
+                    "Failed to get attribute names for AXUIElement: {:?} (error: {}). Wrapping anyway.",
+                    e, err_str
+                );
+                // If we can't get attribute names, be conservative and allow wrapping,
+                // but log for debugging.
+                is_valid = false;
+            }
+        }
```

Comment:
Trying to reduce errors and improve performance on MacOS.
The Apple Accessibility API really kind of sucks 😝  It doesn't really give a great way to introspect so we don't need to make requests which then fail. I opted to get the list of attributes first in an attempt to not need to call `AXRole` if not other attributes can be read. Trying things.